bee-pow/src/providers/miner.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Contains a nonce provider that mine nonces.
+
+use crate::providers::{NonceProvider, NonceProviderBuilder};
+
+use bee_crypto::ternary::{
+    sponge::{BatchHasher, CurlPRounds, BATCH_SIZE},
+    HASH_LENGTH,
+};
+use bee_ternary::{b1t6, Btrit, T1B1Buf, TritBuf};
+
+use crypto::hashes::{blake2b::Blake2b256, Digest};
+use thiserror::Error;
+
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    thread,
+};
+
+const DEFAULT_NUM_WORKERS: usize = 1;
+// Precomputed natural logarithm of 3 for performance reasons.
+// See https://oeis.org/A002391.
+const LN_3: f64 = 1.098_612_288_668_109_8;
```

Comment:
Maybe a silly question, but where does the last `_8` come from? Looking at the decimal expansion the next two digits are `69` which if rounding should go to `7`, not `8`, right?

Reply:
Nice catch, I have no idea how that happened :)

Reply:
Ahhh it was actually clippy, the value with a 7 is too low to be represented as an f64. I just removed it, it's precise enough anyway.